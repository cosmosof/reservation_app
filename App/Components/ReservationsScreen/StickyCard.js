// @flow

import * as React from 'react';
import {View} from 'react-native';

import styles from './Styles/StickyCardStyles';

type Props = { children?: React.Node };

export default function StickyCard(props: Props) {
  return (
    <View style={styles.container}>
      {props.children}
    </View>
  );
}
