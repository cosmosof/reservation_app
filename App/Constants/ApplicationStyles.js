import {Platform} from 'react-native';
import Fonts from './Fonts';
import Colors from './Colors';

const ApplicationStyles = {
  navbarHeader: {
    color: Colors.darkMatBlue2,
    fontSize: 18,
    fontWeight: '600',
    fontFamily: Platform.OS == 'ios' ? Fonts.typeIOS.condensed : Fonts.typeAndroid.condensed,
  },
};

export default {
  ApplicationStyles,
};
