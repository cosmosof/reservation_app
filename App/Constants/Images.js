const images = {
  logo: require('../Images/Relate.png'),
};

export default images;
