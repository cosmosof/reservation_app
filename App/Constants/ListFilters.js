export default [
  {
    index: 0,
    name: 'guest name',
    value: 'name',
  },
  {
    index: 1,
    name: 'in guest names',
    value: 'name_contains',
  },
  {
    index: 3,
    name: 'name starts with',
    value: 'name_starts_with',
  },
  {
    index: 4,
    name: 'hotel name',
    value: 'hotelName',
  },
  {
    index: 5,
    name: 'in hotel names',
    value: 'hotelName_contains',
  },
  {
    index: 7,
    name: 'hotel names start with',
    value: 'hotelName_starts_with',
  },
  {
    index: 9,
    name: 'arrival date',
    value: 'arrivalDate',
  },
  {
    index: 10,
    name: 'departure date',
    value: 'departureDate',
  },
];
