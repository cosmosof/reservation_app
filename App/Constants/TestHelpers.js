const noop = () => {};
const sampleText = 'Sample text';
export default {
  noop,
  sampleText,
};
