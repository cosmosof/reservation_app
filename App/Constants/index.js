import Colors from './Colors';
import Fonts from './Fonts';
import Metrics from './Metrics';
import Images from './Images';
import ListFilters from './ListFilters';
import ApplicationStyles from './ApplicationStyles';
import TestHelpers from './TestHelpers';

export {Colors, Fonts, Images, Metrics, ListFilters, ApplicationStyles, TestHelpers};
