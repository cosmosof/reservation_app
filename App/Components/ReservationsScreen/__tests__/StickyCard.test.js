import React from 'react';
import {shallow} from 'enzyme';
import StickyCard from '../StickyCard';
import {TestHelpers} from '../../../Constants';

const {sampleText} = TestHelpers;

describe('<StickyCard>', () => {
  const Component = <StickyCard />;
  describe('Structure', () => {
    it('renders correctly', () => {
      const wrapper = shallow(Component);
      expect(wrapper).toMatchSnapshot();
    });
    it('renders correctly if children prop is set', () => {
      const Clone = React.cloneElement(Component, {children: sampleText});
      const wrapper = shallow(Clone);
      expect(wrapper).toMatchSnapshot();
    });
  });
});
