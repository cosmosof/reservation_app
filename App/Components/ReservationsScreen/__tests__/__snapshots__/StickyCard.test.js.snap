// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StickyCard> Structure renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "white",
      "flexDirection": "column",
      "justifyContent": "center",
      "marginBottom": 5,
      "paddingTop": 5,
    }
  }
/>
`;

exports[`<StickyCard> Structure renders correctly if children prop is set 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "white",
      "flexDirection": "column",
      "justifyContent": "center",
      "marginBottom": 5,
      "paddingTop": 5,
    }
  }
>
  Sample text
</View>
`;
